packages/api/src/command/medical/patient/convert-fhir-bundle.ts

```diff
@@ -0,0 +1,203 @@
+import { Bundle, Resource } from "@medplum/fhirtypes";
+import { ResourceTypeForConsolidation } from "@metriport/api-sdk";
+import {
+  ConsolidationConversionType,
+  FhirToMedicalRecordPayload,
+} from "@metriport/core/domain/fhir";
+import { logResultToString, makeLambdaClient } from "@metriport/core/external/aws/lambda";
+import { makeS3Client } from "@metriport/core/external/aws/s3";
+import { MetriportError } from "@metriport/core/util/error/metriport-error";
+import { Patient } from "../../../domain/medical/patient";
+import { makeFhirApi } from "../../../external/fhir/api/api-factory";
+import { Config } from "../../../shared/config";
+
+const region = Config.getAWSRegion();
+const lambdaClient = makeLambdaClient(region);
+
+export async function handleBundleToMedicalRecord({
+  bundle,
+  patient,
+  resources,
+  dateFrom,
+  dateTo,
+  conversionType,
+}: {
+  bundle: Bundle<Resource>;
+  patient: Pick<Patient, "id" | "cxId" | "data">;
+  resources?: ResourceTypeForConsolidation[];
+  dateFrom?: string;
+  dateTo?: string;
+  conversionType: ConsolidationConversionType;
+}): Promise<Bundle<Resource>> {
+  const fhir = makeFhirApi(patient.cxId);
+
+  const fhirPatient = await fhir.readResource("Patient", patient.id);
+
+  const bundleWithPatient: Bundle<Resource> = {
+    ...bundle,
+    total: (bundle.total ?? 0) + 1,
+    entry: [
+      {
+        resource: fhirPatient,
+      },
+      ...(bundle.entry ?? []),
+    ],
+  };
+
+  const url = await convertFHIRBundleToMedicalRecord({
+    bundle: bundleWithPatient,
+    patient,
+    resources,
+    dateFrom,
+    dateTo,
+    conversionType,
+  });
+
+  return {
+    resourceType: "Bundle",
+    total: 1,
+    type: "collection",
+    entry: [
+      {
+        resource: {
+          resourceType: "DocumentReference",
+          subject: {
+            reference: `Patient/${patient.id}`,
+          },
+          content: [
+            {
+              attachment: {
+                contentType: `application/${conversionType}`,
+                url: url,
+              },
+            },
+          ],
+        },
+      },
+    ],
+  };
+}
+
+async function convertFHIRBundleToMedicalRecord({
+  bundle,
+  patient,
+  resources,
+  dateFrom,
+  dateTo,
+  conversionType,
+}: {
+  bundle: Bundle<Resource>;
+  patient: Pick<Patient, "id" | "cxId" | "data">;
+  resources?: ResourceTypeForConsolidation[];
+  dateFrom?: string;
+  dateTo?: string;
+  conversionType: ConsolidationConversionType;
+}): Promise<string> {
+  const lambdaName = Config.getFHIRToMedicalRecordLambdaName();
+
+  // create file name
+  const fileName = createFileName({
+    cxId: patient.cxId,
+    patientId: patient.id,
+    resources,
+    dateFrom,
+    dateTo,
+  });
+
+  // store on S3
+  const s3 = makeS3Client(Config.getAWSRegion());
+  const emptyMetaProp = "na";
+  await s3
+    .putObject({
+      Bucket: Config.getMedicalDocumentsBucketName(),
+      Key: fileName,
```

Comment:
just making sure that `putObject` will override if the key already exists?

Reply:
Yep, it does 👍

https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html